eth/filters/filter_system.go

```diff
@@ -351,6 +376,15 @@ func (es *EventSystem) handleTxsEvent(filters filterIndex, ev core.NewTxsEvent)
 	}
 }
 
+func (es *EventSystem) handleFinalizedEvent(filters filterIndex, ev core.ChainEvent) {
+	for _, f := range filters[FinalizedBlockSubscription] {
```

Comment:
Should we check if the hash and block number is empty before sending it to the subscription?
I saw we have a mark to include or not finalize block in `SendBlockEvent` function. In case, the flag `includeFinalized` is false, they might be empty

Reply:
Oh this's my mistake, I have put the condition within `handleFinalizedEvent`. Please have a look!